server/routes/farcast/swap.ts

```diff
@@ -0,0 +1,69 @@
+import router from './index.js'
+import { authMessage, getPageSetting } from './middlewares.js'
+import config from '../../config.js'
+import { lookupFid, renderImageResponse } from '../../src/farcaster.js'
+import type { Request } from 'express'
+
+const generateImageResponse = (text: string, req: Request, retry = false): string => {
+  const host = req.get('host')
+  const image = `${req.protocol}://${host}/${config.farcast.apiBase}/text/image?t=${encodeURIComponent(text)}`
```

Comment:
Can't use dynamic (in the sense that it relies on query parameter) URL for image generation response, because Warpcast will cache the social preview (OG) frame, so the response text becomes stale. You need to create a fixed endpoint for responding image (e.g. `/swap/image`) for serving dynamic content, and set cache policy header in response. See what I did in map extension

Reply:
What about the first frame? Need to redirect to url with random token query param to prevent cache?

Reply:
No, I tried before. Entire frame content is cached. That's why I added image routes in map

Reply:
Image routes in map (`/map/image`) also has query param `token` and generates image from it.
`generateImageResponse()` gets image from text image route with query param `t` and I don't understand what difference `/swap/image` makes from `/text/image`. Would you please help me understand?

Reply:
And another question is if the frame is cached, how can we make the first swap frame
https://github.com/polymorpher/dot-country-embedder/blob/feature/farcast-swap/server/src/farcaster/swap.ts
show updated price?

Reply:
`/map/image` is not used. `/map/stats/image` is the example. Also `/map/image` is fixed per location (token obtained by hashing location string. The frame's content is cached, so the URL to obtain the image is fixed (e.g. `https://foo.bar/stats/image?abc`), but if `https://foo.bar/stats/image?abc` sets cache policy in response header, the image itself will not be cached from the image's URL (if max-age=0). So every time the frame is rendered, a new image is retrieved from the image's URL, even though the frame's content (HTML) itself is cached